src/scripts/main.js

```diff
@@ -1,3 +1,39 @@
 'use strict';
 
 // write your code here
+const logo = document.querySelector('.logo');
+
+// Create promise2 immediately, so it starts rejecting after 2 seconds
+const promise2 = new Promise((resolve, reject) => {
+  setTimeout(() => {
+    reject(new Error('reject'));
+  }, 2000);
+});
+
+// Create promise1 which resolves on logo click
+const promise1 = new Promise((resolve) => {
+  logo.addEventListener('click', () => {
+    resolve('logo clicked');
+  });
+});
+
+// Use Promise.allSettled to handle both
+Promise.allSettled([promise1, promise2]).then((results) => {
+  results.forEach((result) => {
+    if (result.status === 'rejected') {
+      const errorMessage = document.createElement('div');
+
+      errorMessage.classList.add('message', 'error-message');
+      document.body.append(errorMessage);
+      errorMessage.textContent = 'Promise was rejected!';
```

Comment:
The `textContent` should be set before appending the `errorMessage` to the document body to ensure the message is displayed immediately with the correct content.